### PR TITLE
Adding IntervalArray to IntervalRDD

### PR DESCRIPTION
**Pull request by fnothaft**
Continued from #93. Don't merge yet, still testing with https://github.com/bigdatagenomics/adam/pull/1234#issuecomment-259988796.

### CODE REVIEW

File: utils-intervalrdd/src/main/scala/org/bdgenomics/utils/intervalrdd/IntervalPartition.scala [hunk 1]
```diff
@@ -18,21 +18,19 @@
...
+import scala.math.max
 import scala.reflect.ClassTag
 
-protected class IntervalPartition[K <: Interval, V: ClassTag](protected val iTree: IntervalTree[K, V]) extends Serializable with Logging {
+protected class IntervalPartition[K <: Interval[K], V: ClassTag](protected val array: IntervalArray[K, V])
+    extends Serializable with Logging {
 
-  def this() {
-    this(new IntervalTree[K, V]())
+  def getRangeArray(): IntervalArray[K, V] = {
```
Comment (heuermh) [inline utils-intervalrdd/src/main/scala/org/bdgenomics/utils/intervalrdd/IntervalPartition.scala]: missed one, `getRangeArray()` &rarr; `getIntervalArray()`

**Comment by fnothaft** [comment]
Fixed your requested change, @heuermh!

**Comment by fnothaft** [comment]
Still working on integrating with https://github.com/bigdatagenomics/adam/pull/1234, so don't merge yet.

**Comment by fnothaft** [comment]
Merged! Thanks @akmorrow13!

**Pull request merged**